### DISCUSSION TITLE
Testing Stream xread Command with Invalid Stream name/id

### DISCUSSION THREAD

**Discussion by binkabir** [xetorthio/jedis#2810] [Q&A] [answered]
Im trying to test reading a redis stream using the xread command which works as expected. However, when I try to test jedis with a wrong stream name/id it times out the application. 
Is this a bug or an expected result? 
`Response<List<Map.Entry<String, List<StreamEntry>>>>`

**Comment by sazzad16** [comment, marked as answer]
@binkabir What's the value of BLOCK? If it's `0`, then you're getting the expected behavior.

PS: It'd be nicer to know the Jedis version and whole method signature.

**Comment by binkabir** [comment]
Hello @sazzad16 thanks for the response. You are right, the BLOCK value was set to `0` 

I'm using java 16 and  jedis 4.0.1 

```
Map<String, StreamEntryID> stream = Collections.singletonMap("invalid-stream-name", new StreamEntryID());
List<Entry<String, List<StreamEntry>>> streamOutput = jedis.xread(XReadParams.xReadParams().count(1).block(0), stream);
```
